### PR TITLE
SDK-907: Introduced new interfaces, changed namespaces for contracts …

### PR DESCRIPTION
**Pull request by vol4onok**
https://spryker.atlassian.net/browse/SDK-907

### CODE REVIEW

**Review by sergeyspryker — APPROVED** [review]

File: src/Infrastructure/Service/Logger/EventLoggerInterface.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service\Logger;
+
+use SprykerSdk\Sdk\Core\Domain\Event\EventInterface;
+
+interface EventLoggerInterface
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/Logger/EventLoggerInterface.php:12]: This interface breaks layer dependency direction. A lot of services from the Application layer depend on it. Please, move it to the Application layer.

File: src/Infrastructure/Service/ValueReceiver/ReceiverInterface.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver;
+
+use SprykerSdk\Sdk\Core\Application\Dto\ReceiverValueInterface;
+
+interface ReceiverInterface
```
[the remark this answers is not in the document]
Comment (pavelmaksimov25) [reply src/Infrastructure/Service/ValueReceiver/ReceiverInterface.php:12]: First, it is not clear what it does in both cases. We need to name it in a more descriptive way, like `InteractionProcessorInterface`. The second, it doesn't belong to the `Infrastructure` layer, must be in the `Application`.

File: src/Infrastructure/Event/CliRunnerSetupListener.php [hunk 3]
```diff
@@ -16,7 +16,7 @@ class CliRunnerSetupListener
     protected LocalCliRunner $cliRunner;
 
     /**
-     * @param \SprykerSdk\Sdk\Infrastructure\Service\LocalCliRunner $cliRunner
+     * @param \SprykerSdk\Sdk\Infrastructure\Service\CommandRunner\LocalCliRunner $cliRunner
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Event/CliRunnerSetupListener.php:19]: It must be depended on the abstraction. Please change it to the `CommandRunnerInterface` interface.

File: src/Infrastructure/Service/ActionApprover.php [hunk 4]
```diff
@@ -9,18 +9,19 @@
...
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver;
 
 class ActionApprover implements ActionApproverInterface
 {
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/ActionApprover.php:19]: It must be depended on the abstraction. Please change it to the `ReceiverInterface` interface.

File: src/Infrastructure/Service/CommandRunner/CommandRunnerInterface.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service\CommandRunner;
+
+use SprykerSdk\Sdk\Core\Domain\Entity\ContextInterface;
+use SprykerSdk\SdkContracts\Entity\CommandInterface;
+
+interface CommandRunnerInterface
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/CommandRunner/CommandRunnerInterface.php:13]: This one is used in the Application layer. Please move it to the `Application` layer

File: src/Infrastructure/Service/Initializer.php [hunk 6]
```diff
@@ -12,15 +12,16 @@
...
 use SprykerSdk\SdkContracts\Entity\SettingInterface as EntitySettingInterface;
 
 class Initializer implements InitializerInterface
 {
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/Initializer.php:24]: It must be depended on the abstraction. Please change it to the ReceiverInterface interface.

File: src/Infrastructure/Service/ValueReceiver/CliReceiver.php [hunk 7]
```diff
@@ -5,20 +5,19 @@
...
-use SprykerSdk\SdkContracts\ValueReceiver\ValueReceiverInterface;
 use Symfony\Component\Console\Helper\SymfonyQuestionHelper;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Question\ChoiceQuestion;
 use Symfony\Component\Console\Question\ConfirmationQuestion;
 use Symfony\Component\Console\Question\Question;
 
-class CliValueReceiver implements ValueReceiverInterface, InputOutputReceiverInterface
+class CliReceiver implements ReceiverInterface, InputOutputReceiverInterface
```
[the remark this answers is not in the document]
Comment (pavelmaksimov25) [reply src/Infrastructure/Service/ValueReceiver/CliReceiver.php:20]: Already answered in the comment to the `ReceiverInterface`

**Review by Kyrylo-Meretskyi — DISMISSED** [review]

File: src/Infrastructure/Service/WorkflowRunner.php [hunk 8]
```diff
@@ -30,9 +31,9 @@ class WorkflowRunner implements InputOutputReceiverInterface
     protected OutputInterface $output;
 
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/WorkflowRunner.php:36]: It must be depended on the abstraction. Please change it to the `ReceiverInterface` interface.

File: src/Presentation/Console/Command/InitProjectCommand.php [hunk 9]
```diff
@@ -29,9 +29,9 @@ class InitProjectCommand extends Command
     protected const NAME = 'sdk:init:project';
 
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Presentation/Console/Command/InitProjectCommand.php:34]: It must be depended on the abstraction. Please change it to the `ReceiverInterface` interface.

File: src/Presentation/Console/Command/RunWorkflowCommand.php [hunk 10]
```diff
@@ -48,9 +48,9 @@ class RunWorkflowCommand extends Command
     protected ProjectWorkflow $projectWorkflow;
 
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Presentation/Console/Command/RunWorkflowCommand.php:53]: It must be depended on the abstraction. Please change it to the `ReceiverInterface` interface.

File: src/Presentation/Console/Command/ShowWorkflowCommand.php [hunk 11]
```diff
@@ -42,9 +42,9 @@ class ShowWorkflowCommand extends Command
     protected ProjectWorkflow $projectWorkflow;
 
     /**
-     * @var \SprykerSdk\Sdk\Infrastructure\Service\CliValueReceiver
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliReceiver
      */
-    protected CliValueReceiver $cliValueReceiver;
+    protected CliReceiver $cliValueReceiver;
```
Comment (pavelmaksimov25) [inline src/Presentation/Console/Command/ShowWorkflowCommand.php:47]: It must be depended on the abstraction. Please change it to the `ReceiverInterface` interface.

**Review by pavelmaksimov25 — CHANGES_REQUESTED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Review by sergeyspryker — APPROVED** [review]

**Pull request merged**